tests/mapreduce/test.mapreduce.js

```diff
@@ -3697,5 +3697,15 @@ function tests(suiteName, dbName, dbType, viewType) {
         });
       });
     });
+
+    it('should throw when user supplies map func unallowed by strict mode (#6234)', function () {
+      var db = new PouchDB(dbName);
+      return db.query("(function () { globalVar = 'hello'; })()")
```

Comment:
I believe this should be a persisted query rather than a temp view (because that's the code path that actually does an `eval()`), also passing in a string here doesn't work IIRC. You may need to add it to `test.persisted.js` instead, since that's where we keep the persisted tests.